src/plugins/inputs/msg.js

```diff
@@ -19,7 +20,12 @@ module.exports = function(network, chan, cmd, args) {
 		target = chan.name;
 	}
 	var msg = args.join(" ");
-	irc.send(target, msg);
+	var otrSession = client.otrStore.getSession(chan.name, network);
+	if (otrSession && (otrSession.msgstate == OTR.CONST.MSGSTATE_ENCRYPTED)) {
```

Comment:
I see, then I think I'd be happy with a helper on the `OtrStore` side of things to wrap all of these. But here as well it can be done later. :+1: for now with note to move intelligence over there, unless you were planning to do so (i.e. to get rid of your `FIXME`) beforehand. Up to you.